unstructured/ingest/pipeline/pipeline.py

```diff
@@ -40,6 +42,21 @@ def get_nodes_str(self):
         nodes.append(Copier(pipeline_context=self.pipeline_context))
         return " -> ".join([node.__class__.__name__ for node in nodes])
 
+    def expand_batch_docs(self, dict_docs: t.List[dict]) -> t.List[dict]:
+        expanded_docs = []
+        for d in dict_docs:
+            doc = create_ingest_doc_from_dict(d)
+            if isinstance(doc, BaseSingleIngestDoc):
+                expanded_docs.append(doc.to_dict())
+            elif isinstance(doc, BaseIngestDocBatch):
+                expanded_docs.extend([single_doc.to_dict() for single_doc in doc.ingest_docs])
```

Comment:
This is nice. I was expecting it to have to loop through the batch and then loop through single. But this does it all at once.